runtimes/bundles/org.jboss.tools.rsp.server.minishift/src/main/java/org/jboss/tools/rsp/server/minishift/servertype/impl/CRCServerDelegate.java

```diff
@@ -89,10 +89,6 @@ protected IServerStartLauncher getStartLauncher() {
 	@Override
 	public void setDefaults(IServerWorkingCopy server) {
 		server.setAttribute(IMinishiftServerAttributes.LAUNCH_OVERRIDE_BOOLEAN, false);
-		server.setAttribute(IMinishiftServerAttributes.MINISHIFT_CPUS, 4);
```

Comment:
I'm a bit confused why you're setting these "attributes".  Typically we only set attributes we'll need to check later, or attributes we can use to control the server.  Is there a command line flag in crc start where we can pass in the number of cpus we want to  use, or the memory?  If yes, then this delete can be undone, but you need to implement that in the launch controller.  If no, then there's no reason to store these attributes.

Reply:
Yes, there are several flags for the start cmd. 

```
  -b, --bundle string             The system bundle used for deployment of the OpenShift cluster. (default "/home/luca/.crc/crc_libvirt_4.1.14.crcbundle")
  -c, --cpus int                  Number of CPU cores to allocate to the CRC VM (default 4)
  -m, --memory int                MiB of Memory to allocate to the CRC VM (default 8192)
  -n, --nameserver string         Specify nameserver to use for the instance. (i.e. 8.8.8.8)
  -p, --pull-secret-file string   File path of Image pull secret for User (Download it from https://cloud.redhat.com/openshift/install/crc/installer-provisioned)
  -d, --vm-driver string          The driver to use for the CRC VM. Possible values: [libvirt] (default "libvirt")
```

There is the method "getProgramArguments" inside StartCRCLauncher.java (org.jboss.tools.rsp.server.minishift.servertype.impl) which get all these infos. Atleast the cpu, memory and pull secret. I'll check it again tomorrow but it should work.